src/auth/session/storage/custom.ts

```diff
@@ -34,10 +34,19 @@ export class CustomSessionStorage implements SessionStorage {
     }
     if (result) {
       if (result instanceof Session) {
+        if (result.expires && typeof result.expires === 'string') {
+          result.expires = new Date(result.expires);
+        }
+
         return result;
       } else if (result instanceof Object && 'id' in result) {
         let session = new Session(result.id as string);
         session = {...session, ...result};
+
+        if (session.expires && typeof session.expires === 'string') {
+          session.expires = new Date(session.expires);
+        }
```

Comment:
```suggestion
        const session = Session.cloneSession(result, result.id);
```

and following part port to Session.cloneSession.
```ts
        if (session.expires && typeof session.expires === 'string') {
          session.expires = new Date(session.expires);
        }
```

Reply:
Hmm, I don't think that would work - we can't guarantee that `result` will be of type `Session` here, since it may have been loaded from something like a `JSON.parse()` call.

If we tried to call `Session.clone` with `result` it would break in Typescript, and we shouldn't expect the `expires` field to be a string there (it is typed as `Date` so it would have to be set when the session is created anyway).

Reply:
@paulomarg I got it.
My suggestion is for the following reasons:

I wish I could improve the point that users have to convert the `session.expires` to Date type.
(This seems to be solved if `session.isActive()` is implemented)

`session = {...session, ...result};` overwrites `session.id` with `result.id`.
If this assignment is OK, why does `Session.cloneSession` exist?